aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -589,7 +643,9 @@ module aptos_framework::fungible_asset {
 
     /// Decrease the supply of a fungible asset by burning.
     fun decrease_supply<T: key>(metadata: &Object<T>, amount: u64) acquires Supply, ConcurrentSupply {
-        assert!(amount != 0, error::invalid_argument(EAMOUNT_CANNOT_BE_ZERO));
+        if (amount == 0) {
+            return
+        };
         let metadata_address = object::object_address(metadata);
```

Comment:
why?

Reply:
There are several cases where conversion between FA and Coin will happen and there are cases 0 fa or 0 coin may show up. So we should allow 0 in most cases with an early return instead of erroring out.